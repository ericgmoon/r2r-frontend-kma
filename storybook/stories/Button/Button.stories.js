import { action } from "@storybook/addon-actions";
import { text } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react-native";
import React from "react";
import { Text } from "react-native";
import Button from "../../../src/packages/common/button";
import CenterView from "../CenterView";

storiesOf("Button", module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .add("with text", () => <Button>I am a button</Button>)
  .add("with some emoji", () => <Button>😀 😎 👍 💯</Button>);
