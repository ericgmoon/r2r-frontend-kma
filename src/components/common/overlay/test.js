import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";
import Overlay from "./index";
import Button from "../button";

test("should trigger onClosePress", () => {
  const callback = sinon.spy();
  const node = mount(
    <Overlay className="overlay" isVisible showClose onClose={callback}>
      Some text
    </Overlay>
  );
  expect(node.find(".overlay")).toBeDefined();
  node
    .find(".overlay")
    .findWhere((n) => n.prop("testID") === "closeButton")
    .first()
    .props()
    .onPress();
  sinon.assert.called(callback);
});

test("should trigger onPress for main button", () => {
  const callback = sinon.spy();
  const node = mount(
    <Overlay
      className="overlay"
      isVisible
      buttonLabel="Press Me"
      buttonProps={{ onPress: callback }}
    >
      Button
    </Overlay>
  );
  expect(node.find(".overlay")).toBeDefined();
  node
    .find(".overlay")
    .findWhere((n) => n.prop("testID") === "mainOverlayButton")
    .first()
    .props()
    .onPress();
  sinon.assert.called(callback);
});

describe("renders correctly", () => {
  test("without props", () => {
    const node = renderer.create(<Overlay isVisible />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with title", () => {
    const node = renderer.create(<Overlay isVisible title="Title" />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with close icon", () => {
    const node = renderer
      .create(<Overlay isVisible isCloseIconVisible />)
      .toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with custom button", () => {
    const node = renderer
      .create(<Overlay isVisible buttonLabel="Press Me" />)
      .toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with text", () => {
    const node = renderer.create(<Overlay isVisible>Button</Overlay>).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with button as child", () => {
    const node = renderer
      .create(
        <Overlay isVisible>
          <Button>Press Me</Button>
        </Overlay>
      )
      .toJSON();
    expect(node).toMatchSnapshot();
  });
});
