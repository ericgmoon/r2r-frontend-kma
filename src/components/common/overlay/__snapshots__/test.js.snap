// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with button as child 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  theme={
    Object {
      "colors": Object {
        "black": "#242424",
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "grey": "rgba(0, 0, 0, 0.54)",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "default": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "ios": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "web": Object {
            "error": "#ff190c",
            "grey": "#393e42",
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#ca71eb",
        "success": "#52c41a",
        "warning": "#faad14",
        "white": "#ffffff",
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 20,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <View />
          <View
            style={
              Object {
                "height": 16,
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": 36,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "background": "#000",
                    "backgroundColor": "black",
                    "borderColor": "black",
                    "borderRadius": 36,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 16,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  PRESS ME
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly with close icon 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  theme={
    Object {
      "colors": Object {
        "black": "#242424",
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "grey": "rgba(0, 0, 0, 0.54)",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "default": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "ios": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "web": Object {
            "error": "#ff190c",
            "grey": "#393e42",
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#ca71eb",
        "success": "#52c41a",
        "warning": "#faad14",
        "white": "#ffffff",
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 20,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <View />
          <View
            style={
              Object {
                "height": 16,
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly with custom button 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  theme={
    Object {
      "colors": Object {
        "black": "#242424",
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "grey": "rgba(0, 0, 0, 0.54)",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "default": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "ios": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "web": Object {
            "error": "#ff190c",
            "grey": "#393e42",
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#ca71eb",
        "success": "#52c41a",
        "warning": "#faad14",
        "white": "#ffffff",
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 20,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <View />
          <View
            style={
              Object {
                "height": 16,
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "borderRadius": 36,
                "overflow": "hidden",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "busy": false,
                  "disabled": false,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "background": "#000",
                    "backgroundColor": "black",
                    "borderColor": "black",
                    "borderRadius": 36,
                    "borderWidth": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "padding": 8,
                    "paddingBottom": 16,
                    "paddingLeft": 20,
                    "paddingRight": 20,
                    "paddingTop": 16,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 18,
                      "paddingVertical": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  PRESS ME
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly with text 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  theme={
    Object {
      "colors": Object {
        "black": "#242424",
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "grey": "rgba(0, 0, 0, 0.54)",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "default": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "ios": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "web": Object {
            "error": "#ff190c",
            "grey": "#393e42",
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#ca71eb",
        "success": "#52c41a",
        "warning": "#faad14",
        "white": "#ffffff",
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 20,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <View />
          <View
            style={
              Object {
                "height": 16,
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        >
          Button
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly with title 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  theme={
    Object {
      "colors": Object {
        "black": "#242424",
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "grey": "rgba(0, 0, 0, 0.54)",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "default": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "ios": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "web": Object {
            "error": "#ff190c",
            "grey": "#393e42",
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#ca71eb",
        "success": "#52c41a",
        "warning": "#faad14",
        "white": "#ffffff",
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 20,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <View>
            <Text
              style={
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Title
            </Text>
          </View>
          <View
            style={
              Object {
                "height": 16,
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;

exports[`renders correctly without props 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  theme={
    Object {
      "colors": Object {
        "black": "#242424",
        "disabled": "hsl(208, 8%, 90%)",
        "divider": "#bcbbc1",
        "error": "#ff190c",
        "grey0": "#393e42",
        "grey1": "#43484d",
        "grey2": "#5e6977",
        "grey3": "#86939e",
        "grey4": "#bdc6cf",
        "grey5": "#e1e8ee",
        "greyOutline": "#bbb",
        "platform": Object {
          "android": Object {
            "error": "#f44336",
            "grey": "rgba(0, 0, 0, 0.54)",
            "primary": "#2196f3",
            "secondary": "#9C27B0",
            "success": "#4caf50",
            "warning": "#ffeb3b",
          },
          "default": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "ios": Object {
            "error": "#ff3b30",
            "grey": "#7d7d7d",
            "primary": "#007aff",
            "searchBg": "#dcdce1",
            "secondary": "#5856d6",
            "success": "#4cd964",
            "warning": "#ffcc00",
          },
          "web": Object {
            "error": "#ff190c",
            "grey": "#393e42",
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#ca71eb",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        },
        "primary": "#2089dc",
        "searchBg": "#303337",
        "secondary": "#ca71eb",
        "success": "#52c41a",
        "warning": "#faad14",
        "white": "#ffffff",
      },
    }
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .4)",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
    testID="RNE__Overlay__backdrop"
  />
  <View
    pointerEvents="box-none"
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderRadius": 20,
          "padding": 20,
          "shadowColor": "rgba(0, 0, 0, .3)",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowRadius": 4,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "height": 24,
              "justifyContent": "space-between",
              "marginBottom": 20,
            }
          }
        >
          <View />
          <View
            style={
              Object {
                "height": 16,
                "width": 24,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "marginBottom": 20,
              "marginTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</Modal>
`;
